Documentation/api-guidelines/nullability.md

```diff
@@ -0,0 +1,92 @@
+2. Thousands upon thousands of types we don't control override this method today.
+3. It's common for helper routines to invoke via the base object.ToString, but many ToString uses are actually on derived types.  This is particularly true when working in a code base that both defines a type and consumes its ToString.
+4. Based on examination of several large code bases, we believe it to be relatively rare that the result of an Object.ToString call (made on the base) to be directly dereferenced.  It's much more common to pass it to another method that accepts `string?`, such as `String.Concat`, `String.Format`, `Console.WriteLine`, logging utilities, and so on.  And while we advocate that ToString results shouldn't be assumed to be in a particular machine-readable format and parsed, it's certainly the case that code bases do, such as using `Substring` on the result, but in such cases, the caller needs to understand the format of what's being rendered, which generally means they're working with a derived type rather than calling through the base Object.ToString.
+
+As such, for now, we will start with `Object.ToString` returning `string?`.  We can re-evaluate this decision as we get more experience with consumers of the feature.
+
+In contrast, for `Exception.Message` which is also virtual, we plan to have it be non-nullable, even though technically a derived class could override it to return null, because doing so is so rare that we couldn't find any meaningful examples of doing so.
+
+## Code Review Guidance
+
+Code reviews for enabling the nullability warnings are particularly interesting in that often differ significantly from general code reviews.  Typically, a code reviewer focused only on the code actually being modified, and thus that shows up in a code review diffing tool.  However, enabling the nullability feature has a much broader impact, in that it effectively inverts the meaning of every reference type use in the codebase (or, more specifically, in the scope at which the nullability warning context was applied).  So, for example, if you turn on nullability for a whole file (`#enable nullable` at the top of the file) and then touch no other lines in the file, every method that accepts a `string` is now accepting a non-nullable `string`; whereas previously passing in `null` to that argument would be fine, now the compiler will warn about it, and to allow nulls, the argument must be changed to `string?`.  This means that enabling nullability checking requires reviewing all exposed APIs in that context, regardless of whether they were modified or not, as the contract exposed by the API may have been implicitly modified.
```

Comment:
> Typically, a code reviewer focused only on the code actually being modified, and thus that shows up in a code review diffing tool.

I don't understand.

Reply:
> I don't understand.

When you do a code review, do you review only what's highlighted by the diffing tool, or do you review the entire thousands of lines of code in the file that weren't touched?

Reply:
It was difficult to parse the "..., and thus that shows up...".

How about:
Typically, a code reviewer **focuses** only on the code actually being modified (i.e. the lines highlighted by the code review diffing tool).